doc/papers/joss/paper.md

```diff
@@ -0,0 +1,115 @@
+---
+title: 'libCEED: An open-source library for efficient high-order operator evaluations'
+tags:
+  - high-performance computing
+  - high-order methods
+  - finite elements
+  - spectral elements
+  - matrix-free
+authors:
+  - name: Jed Brown
+    orcid: 0000-0002-9945-0639
+    affiliation: 1
+  - name: Jeremy Thompson
+    affiliation: 1
+  - name: Valeria Barra
+    orcid: 0000-0003-1129-2056
+    affiliation: 1
+  - name: XXXX TODO Many more authors to be added XXXX
+    affiliation: 2
+affiliations:
+ - name: University of Colorado Boulder
+   index: 1
+ - name: Lawrence Livermore National Laboratory
+   index: 2
+date: XXXX TODO write date XXXX
+bibliography: paper.bib
+
+# Optional fields if submitting to a AAS journal too, see this blog post:
+# https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
+aas-doi: 10.3847/xxxxx <- update this with the DOI from AAS once you know it.
+aas-journal: Astrophysical Journal <- The name of the AAS journal.
+---
+
+# Summary
+
+High-order numerical methods are widely used in Partial Differential Equation (PDE)
+solvers, but software packages that provide high-performance implementations have
+often been special-purpose and intrusive. ``libCEED`` is an open-source light-weight
+library that offers a purely algebraic interface for efficient operator evaluation
+and matrix-free preconditioning ingredients. libCEED supports run-time selection of
+implementations tuned for a variety of computational architectures, including CPUs and
+GPUs, and can be unobtrusively integrated in new and legacy software to provide portable
+performance. We introduce ``libCEED``’s conceptual framework and interface, and show examples of its integration with other packages, including PETSc and MFEM.
+
+In finite element formulations, the weak form of a PDE is evaluated on a subdomain
+(element), and the local results are composed into a larger system of equations that
+models the entire problem. In particular, when high-order finite elements or spectral
+elements are used, the resulting sparse matrix representation of the global operator
+is computationally expensive, with respect to both the memory transfer and floating
+point operations needed for its evaluation [@Brown:2010]. ``libCEED`` provides an
+interface for matrix-free operator description that enables efficient evaluation on
+a variety of computational device types (selectable at run time).
+
+# libCEED's API
+
+``libCEED``'s Application Programming Interface (API) provides the local action of the
```

Comment:
Do we want to mention C99/F77/Python interface. Not sure if we want to direct people to the binder?

Reply:
Yeah, this was one of my questions on Zulip. It is up to us to decide. I would totally mention the interfaces. To what extent to describe them is totally up to us. If we want to describe the Python one a bit better, I have some paragraphs ready that I wrote for the SciPy submission

Reply:
Done in [d6f5a2d](https://github.com/CEED/libCEED/pull/530/commits/d6f5a2d9c1e6f05447ee70f32025e58102c9e145)